src/testRunner/unittests/tscWatch/watchEnvironment.ts

```diff
@@ -497,5 +497,161 @@ namespace ts.tscWatch {
                 verifyWorker("-extendedDiagnostics");
             });
         });
+
+        describe("handles watch for extended configs", () => {
+            verifyTscWatch({
+                scenario,
+                subScenario: "watchExtends/with several extended configs",
+                commandLineArgs: ["-w", "-p", "/a/b/tsconfig.json"],
+                sys: () => {
+                    const firstExtendedFile: File = {
+                        path: "/a/b/first.tsconfig.json",
+                        content: JSON.stringify({
+                            compilerOptions: {
+                                strict: false,
+                            }
+                        })
+                    };
+                    const secondExtendedFile: File = {
+                        path: "/a/b/second.tsconfig.json",
+                        content: JSON.stringify({
+                            extends: "./first.tsconfig.json",
+                            compilerOptions: {
+                                strictNullChecks: true,
+                            }
+                        })
+                    };
+                    const thirdExtendedFile: File = {
+                        path: "/a/b/third.tsconfig.json",
+                        content: JSON.stringify({
+                            extends: "./second.tsconfig.json",
+                            compilerOptions: {
+                                strictBindCallApply: true,
+                            }
+                        })
+                    };
+                    const fourthExtendedFile: File = {
+                        path: "/a/b/fourth.tsconfig.json",
+                        content: JSON.stringify({
+                            extends: "./third.tsconfig.json",
+                            compilerOptions: {
+                                strictFunctionTypes: true,
+                            }
+                        })
+                    };
+                    const configFile: File = {
+                        path: "/a/b/tsconfig.json",
+                        content: JSON.stringify({
+                            extends: "./fourth.tsconfig.json",
+                            compilerOptions: {
+                                strictPropertyInitialization: true,
+                            }
+                        })
+                    };
+                    const files = [
+                        libFile, commonFile1, commonFile2, firstExtendedFile, secondExtendedFile, thirdExtendedFile,
+                        fourthExtendedFile, configFile
+                    ];
+                    return createWatchedSystem(files);
+                },
+                changes: emptyArray

```

Comment:
You want changes to be set and verify that change to extended file results in rebuilding program and updates the program